src/interfaces/config.ts

```diff
@@ -24,6 +26,51 @@ export type VersionDetails = {
   shell?: string
 }
 
+export interface Theme {
+  alias?: Color
+  bin?: Color
+  command?: Color
+  commandSummary?: Color
+  dollarSign?: Color
+  flag?: Color
+  flagDefaultValue?: Color
+  flagOptions?: Color
+  flagRequired?: Color
+  flagSeparator?: Color
+  flagType?: Color
+  sectionDescription?: Color
+  sectionHeader?: Color
+  topic?: Color
+  version?: Color
+}
+
+// TODO: find a way to create this array at build time based on the interface keys
+export const THEME_KEYS = [
+  'alias',
+  'bin',
+  'command',
+  'commandSummary',
+  'dollarSign',
+  'flag',
+  'flagDefaultValue',
+  'flagOptions',
+  'flagRequired',
+  'flagSeparator',
+  'flagType',
+  'sectionDescription',
+  'sectionHeader',
+  'topic',
+  'version',
+]
+
+export function parseTheme(untypedTheme: Record<string, string>): Theme {
+  return Object.fromEntries(
+    Object.entries(untypedTheme)
+      .filter(([key]) => THEME_KEYS.includes(key))
```

Comment:
I don't understand the value of this line. I understand that it's going to remove any un-supported theme key from the final result but why would it be bad to have unsupported keys in the final theme? They won't do any harm. It also would allow a future where the theme could be customized beyond just help. For instance, someone wants to have a theme for their CLI's log messages

Plus, if you remove it then you can remove the `THEME_KEYS` const that just duplicates the `Theme` type

But if you think it's absolutely imperative to filter the keys, I think this is a cleaner solution:

```typescript
export const THEME_KEYS = [
  'alias',
  'bin',
  'command',
  'commandSummary',
  'dollarSign',
  'flag',
  'flagDefaultValue',
  'flagOptions',
  'flagRequired',
  'flagSeparator',
  'flagType',
  'sectionDescription',
  'sectionHeader',
  'topic',
  'version',
]

export type Theme = Record<string, Color>

export function parseTheme(untypedTheme: Record<string, string>): Theme {
  return Object.fromEntries(
    Object.entries(untypedTheme)
      .filter(([key]) => THEME_KEYS.includes(key))
      // eslint-disable-next-line new-cap
      .map(([key, value]) => [key, new Color.default(value)]),
  )
}
```

Reply:
I will keep the filter to: 

- avoid wasting time parsing color codes that are never going to be used
- avoid storing colors in memory that are never going to be used 

I know it does not make any difference in performance since it is a really small number of props, but I want to leave it there anyway.

<img width="627" alt="image" src="https://github.com/oclif/core/assets/55927613/45fc6edc-7e83-4fb8-bb11-e1f85a21ebd0">

Reply:
fixed